tfx/components/transform/executor_test.py

```diff
@@ -191,10 +194,45 @@ def testDoWithDuplicatePreprocessingFn(self):
       self._transform_executor.Do(self._input_dict, self._output_dict,
                                   self._exec_properties)
 
+  def testDoWithCustomSplits(self):
+    self._exec_properties['splits_config'] = json_format.MessageToJson(
+        transform_pb2.SplitsConfig(analyze_splits=['train'],
+                                   transform_splits=['eval']),
+        preserving_proto_field_name=True)
+    self._exec_properties['module_file'] = self._module_file
+    self._transformed_examples.split_names = artifact_utils.encode_split_names(
+        ['eval'])
+    self._output_dict[executor.TRANSFORMED_EXAMPLES_KEY] = [
+        self._transformed_examples]
+
+    self._transform_executor.Do(self._input_dict, self._output_dict,
+                                self._exec_properties)
+    self.assertFalse(
+        tf.io.gfile.exists(
+            os.path.join(self._transformed_examples.uri, 'train')))
+    self.assertNotEqual(
+        0,
+        len(
+            tf.io.gfile.listdir(
+                os.path.join(self._transformed_examples.uri, 'eval'))))
+    path_to_saved_model = os.path.join(
+        self._transformed_output.uri, tft.TFTransformOutput.TRANSFORM_FN_DIR,
+        tf.saved_model.SAVED_MODEL_FILENAME_PB)
+    self.assertTrue(tf.io.gfile.exists(path_to_saved_model))
+
+  def testDoWithEmptyAnalyzeSplits(self):
```

Comment:
Should we also have a test case of empty transform_splits?

Reply:
i think empty transform_splits is not a special case, it's equivalent to what is tested in testDoWithCustomSplits

i can still add a test for empty transform_splits if you think it's necessary and more clear